src/qibolab/_core/backends.py

```diff
@@ -98,6 +126,10 @@ def execute_circuit(self, circuit, initial_state=None, nshots=1000):
                 "Hardware backend only supports circuits as initial states.",
             )
 
+        # Temporary fix: overwrite the wire names
+        if not all(q in self.qubits for q in circuit.wire_names):
+            circuit.wire_names = self.qubits[: circuit.nqubits]
```

Comment:
In principle, this was temporary for a certain version of Qibo. But now this should be over since enough versions, so we may just get rid of this (but I need to verify first in Qibo)

Reply:
> In principle, this was temporary for a certain version of Qibo. But now this should be over since enough versions, so we may just get rid of this (but I need to verify first in Qibo)

Do you mean the fact that we were using `circuit._wire_names` instead of the actual setter (`circuit.wire_names`) or having these lines altogether. I think the overwrite is needed, because the qibo circuit created by the user may not match the qubit names we are executing on (eg. think of platforms with non-integer qubit names). However, if this overwrite is already done on the qibo side, which is possible since we are now exposing the qubit names from the backend, then here it is duplicated and we can drop.